### DISCUSSION TITLE
Does anyone know the origin of the algorithms  used in JSCAD?

### DISCUSSION THREAD

**Discussion by briansturgill** [jscad/OpenJSCAD.org#1097] [General]
I'm wondering what text or papers were used.
I am especially interested in the boolean operations -- from reading I am pretty sure that they should run faster than they do.
Anyway, I want to see what the original algorithm said before I try other paths.

Thanks!

**Comment by z3dev** [comment]
As the readme says...

Early CSG Library by:

- Evan Wallace,
- Eduard Bespalov,
- Joost Nieuwenhuijse,
- Alexandre Girard

**Reply by briansturgill** [reply]
@z3dev No, I've seen the author list.. what I mean is where were they taking the algorithms from?
For instance "Computational Geometry in C" by Joseph O'Rourke would like have been popular at the time.
I want to understand the algorithm behind the booleans and the source code isn't sufficient for my purposes.

**Reply by z3dev** [reply]
No idea. You can try asking Evan Wallace

**Reply by briansturgill** [reply]
@z3dev Good idea, I'll see if I can find email for him.

**Comment by hrgdavor** [comment]
this is closest I have found https://en.wikipedia.org/wiki/Binary_space_partitioning

**Comment by thehans** [comment]
See here: https://github.com/evanw/csg.js/#implementation-details
Plus the comments in the source of that repo (not sure how much if any of that remains in openjscad today).

**Reply by hrgdavor** [reply]
I am becoming a big fan of https://github.com/evanw
I am enjoying jscad that has csg.js forked for years, and also love Figma ... and coincidentally my current favorite js build tool is esbuild !!!!
